pandas/tests/tslibs/test_strptime.py

```diff
@@ -59,3 +59,39 @@ def test_array_strptime_resolution_mixed(self, tz):
         fmt = "ISO8601"
         res, _ = array_strptime(arr, fmt=fmt, utc=False, creso=creso_infer)
         tm.assert_numpy_array_equal(res, expected)
+
+    def test_array_strptime_resolution_todaynow(self):
+        # specifically case where today/now is the *first* item
+        vals = np.array(["today", np.datetime64("2017-01-01", "us")], dtype=object)
+
+        now = Timestamp("now").asm8
+        res, _ = array_strptime(vals, fmt="%Y-%m-%d", utc=False, creso=creso_infer)
+        res2, _ = array_strptime(
+            vals[::-1], fmt="%Y-%m-%d", utc=False, creso=creso_infer
+        )
+
+        # 1ms is an arbitrary cutoff for call overhead; in local testing the
+        #  actual difference is about 250us
+        tolerance = np.timedelta64(1, "ms")
+
+        assert res.dtype == "M8[us]"
+        assert abs(res[0] - now) < tolerance
```

Comment:
I would be OK skipping the comparison to a separate "now" to potentially avoid near-miss occasional failures. I think at least passing `"today"` and `"now"` through `array_strptime` and seeing if `us` is return is good enough

Reply:
would you be happier with a bigger tolerance?  I can imagine regressions here that cause this to miss by a huge amount, hard to imagine regressions that miss by a small amount

Reply:
Yeah a larger tolerance would give me more peace.